### PR TITLE
`KedroDataCatalog`  release updates

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Description
Context: https://github.com/orgs/kedro-org/projects/2/views/4?pane=issue&itemId=81553599

## Development notes

1. Updated release notes, module and class docstrings with the clarification about future possible breaking changes and added usage example.
2. Removed previously added `add_data()` method as it was considered as odd when working on https://github.com/kedro-org/kedro/issues/4175, https://github.com/kedro-org/kedro/blob/643219d92400bc34fdcec1081609b751d88afe8e/kedro/io/kedro_data_catalog.py#L338
So it was decided not to introduce it.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by lrcouto — APPROVED** [review]
LGTM!

File: RELEASE.md [hunk 1]
```diff
@@ -4,20 +4,22 @@
...
+  * `add_feed_dict()` was simplified to only add raw data;
   * Datasets' initialisation was moved out from `from_config()` method to the constructor.
 * Moved development requirements from `requirements.txt` to the dedicated section in `pyproject.toml` for project template.
 * Implemented `Protocol` abstraction for the current `DataCatalog` and adding new catalog implementations.
 * Refactored `kedro run` and `kedro catalog` commands.
 * Moved pattern resolution logic from `DataCatalog` to a separate component - `CatalogConfigResolver`. Updated `DataCatalog` to use `CatalogConfigResolver` internally.
 * Made packaged Kedro projects return `session.run()` output to be used when running it in the interactive environment.
 * Enhanced `OmegaConfigLoader` configuration validation to detect duplicate keys at all parameter levels, ensuring comprehensive nested key checking.
+
+**Note:** ``KedroDataCatalog`` is an experimental feature, so please mind possible breaking changes while using it.
```
Comment (merelcht) [inline RELEASE.md:16]:
```suggestion
**Note:** ``KedroDataCatalog`` is an experimental feature and is under active development. Therefore, it is possible we'll introduce breaking changes to this class, so be mindful of that if you decide to use it already. Let us know if you have any feedback about the ``KedroDataCatalog`` or ideas for new features.
```

File: kedro/io/kedro_data_catalog.py [hunk 2]
```diff
@@ -44,6 +47,9 @@ def __init__(
         single point of reference for your calls, relaying load and save
         functions to the underlying datasets.
 
+        Note: ``KedroDataCatalog`` is an experimental feature, so please mind
+        possible breaking changes while using it.
```
Comment (merelcht) [inline kedro/io/kedro_data_catalog.py:51]:
```suggestion
        Note: ``KedroDataCatalog`` is an experimental feature and is under active development. Therefore, it is possible we'll introduce breaking changes to this class, so be mindful of that if you decide to use it already.
```

**Review by merelcht — APPROVED** [review]
Some small adjustments to wording, but otherwise looks good 👍

**Pull request merged**